### ISSUE TITLE
Crashing Issue

### ISSUE THREAD

**Issue by HarryBrew69** [N3MIS15/ESP32-Tilt-Repeater#1]
Hi Mate, I recently checked on the repeater devices I have (8 of them) as you do. I noticed that they were not working so I set about reflashing them. Anyway, upon reboot, the ESP was throwing the error code which I have pasted below. Needless to say, they are no longer repeating. Any ideas? 
Cheers
Harry.  

1 Devices Found.
Device #0 is a Tilt
--------------------------------
Colour: Pink
Temp: 23.33C
Gravity: 1.072
a495bb80c5b14b44b5121370f02d74de004a043055
--------------------------------
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x4000baad  PS      : 0x00060130  A0      : 0x80191518  A1      : 0x3ffc9420  
A2      : 0x3ffc46c8  A3      : 0x3ffc9450  A4      : 0x00000002  A5      : 0x00000001  
A6      : 0x00060120  A7      : 0x00000000  A8      : 0x80192216  A9      : 0x0000000b  
A10     : 0x00000002  A11     : 0x00000000  A12     : 0x8008e764  A13     : 0x3ffbc450  
A14     : 0x00000000  A15     : 0x3ffc77c0  SAR     : 0x0000001e  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000001  LBEG    : 0x400014fd  LEND    : 0x4000150d  LCOUNT  : 0xffffffff  

Backtrace: 0x4000baad:0x3ffc9420 0x40191515:0x3ffc9450 0x400fae2e:0x3ffc9480 0x400d1fc7:0x3ffc94a0 0x400d2452:0x3ffc95b0 0x400d72eb:0x3ffc9700 0x4008e551:0x3ffc9720

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:1216
ho 0 tail 12 room 4
load:0x40078000,len:9720
ho 0 tail 12 room 4
load:0x40080400,len:6352
entry 0x400806b8

**Comment by N3MIS15** [comment]
Yep, looks to be a bug. Unfortunately I'm stuck at work for the next 8 hours and will have to look it after that. Before I try to decode that backtrace, is the code you are using current master?

**Comment by N3MIS15** [comment]
Hi Harry. I have been trying to figure out whats going on here. 
I am unable to replicate a crash unless I downgrade the esp boards definitions.

Can you please let me know which version you have. 
In Arduino, go to Tools -> Board -> Boards Manager. In there search for esp32.
You should now have the following:

```
**esp32**
by **Espressif Systems** version **1.0.4** INSTALLED
Boards included in this package:
ESP32 Dev Module, WEMOS LoLin32, WEMOS D1 MINI ESP32.
```
and so on..

The code wouldn't compile with version 1.0.0/1.0.1. It does compile with versions 1.0.2/1.0.3 but scanning results in a crash similar to yours.

The crash appears to be in the bluetooth handling code rather than my own (although I am not ruling that out) Anyway, if you do not have version 1.0.4 installed, please do so and test again.

Cheers.

**Comment by HarryBrew69** [comment]
Hi mate,
Yes I have the 1.0.4 installed. Still struggling with it. 
here is the readout from the upload. 

_"Sketch uses 1097129 bytes (83%) of program storage space. Maximum is 1310720 bytes.
Global variables use 39968 bytes (12%) of dynamic memory, leaving 287712 bytes for local variables. Maximum is 327680 bytes.
esptool.py v2.6
Serial port COM3
Connecting........_____....._____.
Chip is ESP32D0WDQ6 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
MAC: 24:0a:c4:c6:46:b4
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Compressed 8192 bytes to 47...
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.1 seconds (effective 736.4 kbit/s)...
Hash of data verified.
Compressed 15856 bytes to 10276...
Wrote 15856 bytes (10276 compressed) at 0x00001000 in 0.2 seconds (effective 598.3 kbit/s)...
Hash of data verified.
Compressed 1097360 bytes to 612608...
Wrote 1097360 bytes (612608 compressed) at 0x00010000 in 11.9 seconds (effective 737.0 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 128...
Wrote 3072 bytes (128 compressed) at 0x00008000 in 0.1 seconds (effective 253.4 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin..."_

And here is the error codes, I notice that the first error is different from the second.
"Colour: Pink
Temp: 21.11C
Gravity: 1.086
a495bb80c5b14b44b5121370f02d74de0046043e55
--------------------------------
Guru Meditation Error: Core  1 panic'ed (InstrFetchProhibited). Exception was unhandled.
Core 1 register dump:
PC      : 0x00000000  PS      : 0x00060130  A0      : 0x80191518  A1      : 0x3ffc9420  
A2      : 0x3ffc46c8  A3      : 0x3ffc9450  A4      : 0x00000002  A5      : 0x00000001  
A6      : 0x00060120  A7      : 0x00000000  A8      : 0x80192216  A9      : 0x0000000b  
A10     : 0x00000002  A11     : 0x00000000  A12     : 0x8008e764  A13     : 0x3ffbc450  
A14     : 0x00000000  A15     : 0x3ffc77c0  SAR     : 0x0000001c  EXCCAUSE: 0x00000014  
EXCVADDR: 0x00000000  LBEG    : 0x400014fd  LEND    : 0x4000150d  LCOUNT  : 0xffffffff  

Backtrace: 0x00000000:0x3ffc9420 0x40191515:0x3ffc9450 0x400fae2e:0x3ffc9480 0x400d1fc7:0x3ffc94a0 0x400d2452:0x3ffc95b0 0x400d72eb:0x3ffc9700 0x4008e551:0x3ffc9720

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:8896
load:0x40080400,len:5816
entry 0x400806ac

Scanning...
1 Devices Found.
Device #0 is a Tilt
--------------------------------
Colour: Pink
Temp: 21.11C
Gravity: 1.086
a495bb80c5b14b44b5121370f02d74de0046043e55
--------------------------------
Guru Meditation Error: Core  1 panic'ed (IllegalInstruction). Exception was unhandled.
Core 1 register dump:
PC      : 0x62343462  PS      : 0x00060730  A0      : 0x80191518  A1      : 0x3ffc9420  
A2      : 0x3ffc46c8  A3      : 0x62343462  A4      : 0x3ffc4710  A5      : 0x00000001  
A6      : 0x00060120  A7      : 0x00000000  A8      : 0x80192250  A9      : 0x3ffc9410  
A10     : 0x00000000  A11     : 0x3ffc9420  A12     : 0xffffffff  A13     : 0x3ffbbc76  
A14     : 0x0000070b  A15     : 0x00000000  SAR     : 0x0000001c  EXCCAUSE: 0x00000000  
EXCVADDR: 0x00000000  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0xffffffff  

Backtrace: 0x62343462:0x3ffc9420 0x40191515:0x3ffc9450 0x400fae2e:0x3ffc9480 0x400d1fc7:0x3ffc94a0 0x400d2452:0x3ffc95b0 0x400d72eb:0x3ffc9700 0x4008e551:0x3ffc9720

Rebooting...
ets Jun  8 2016 00:22:57
"
I am happy to keep trying if you need any more info just say so.
Cheers
harry

**Comment by HarryBrew69** [comment]
So I couldn't leave it alone. I uninstalled arduino IDE and went through my PC looking for remaining files. I found an old folder that had library files in it. I deleted it and installed the arduino IDE and ran the upload again and bingo. I guess it must have been indexing the old filed regardless of the fact that the new files were there also. Anyway, a lesson in keeping you PC free of old files. Thanks for the help buddy.
